base/strings/string.jl

```diff
@@ -34,46 +34,35 @@ const utf8_trailing = [
 ## required core functionality ##
 
 function endof(s::String)
-    d = s.data
-    i = length(d)
-    @inbounds while i > 0 && is_valid_continuation(d[i])
+    p = pointer(s)
```

Comment:
Any idea what would be a good api? I'm mainly not sure about how to express the scope a `gcroot` applies to.....